File: seed/csharp-model/trace/src/SeedTrace/Submission/InitializeProblemRequest.cs
using System.Text.Json.Serialization;
using SeedTrace.Core;

namespace SeedTrace;

public record InitializeProblemRequest
{
    [JsonPropertyName("problemId")]
    public required string ProblemId { get; set; }

    [JsonPropertyName("problemVersion")]
    public int? ProblemVersion { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-model/trace/src/Traits/InitializeProblemRequest.php
<?php

namespace Seed\Traits;

use Seed\Core\Json\JsonProperty;

/**
 * @property string $problemId
 * @property ?int $problemVersion
 */
trait InitializeProblemRequest
{
    /**
     * @var string $problemId
     */
    #[JsonProperty('problemId')]
    public string $problemId;

    /**
     * @var ?int $problemVersion
     */
    #[JsonProperty('problemVersion')]
    public ?int $problemVersion;
}
